bestiary/core/models.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2014-2020 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Authors:
+#     Santiago Dueñas <sduenas@bitergia.com>
+#     Miguel Ángel Fernández <mafesan@bitergia.com>
+#
+
+import datetime
+
+from dateutil import tz
+
+from django.db.models import (CASCADE,
+                              Model,
+                              BooleanField,
+                              CharField,
+                              DateTimeField,
+                              ForeignKey)
+
+from django_mysql.models import JSONField
+
+from enum import Enum
+
+# Default dates for periods
+MIN_PERIOD_DATE = datetime.datetime(1900, 1, 1, 0, 0, 0,
+                                    tzinfo=tz.tzutc())
+MAX_PERIOD_DATE = datetime.datetime(2100, 1, 1, 0, 0, 0,
+                                    tzinfo=tz.tzutc())
+
+# Innodb and utf8mb4 can only index 191 characters
+# For more information regarding this topic see:
+# https://dev.mysql.com/doc/refman/5.5/en/charset-unicode-conversion.html
+MAX_SIZE_CHAR_INDEX = 191
+MAX_SIZE_CHAR_FIELD = 128
+
+
+class Transaction(Model):
+    """Model class for Transaction objects
+
+    This class is meant to represent objects which are
+    created inside a context (e.g., a method from the API)
+    where the database is going to be modified in some way.
+
+    Every transaction object must have a unique identifier
+    (`tuid`) and the name of the method creating the
+    transaction (`name`).
+
+    Attributes:
+    * tuid`: Unique identifier for the transaction.
+    * name`: Name of the method creating the transaction.
+    * created_at`: Datetime when the transaction is created.
+    * closed_at`: Datetime when the transaction is closed.
+    * is_closed`: Boolean value, `True` if the transaction is closed.
+    * authored_by`: Username from who is creating the transaction.
+    """
```

Comment:
I was not sure if this was the right way to document Model classes. @sduenas What do you think about the format above?